File: sample-projects/objectivec/CellarsAndCentaurs/Application/CreatureUtil.m
@import Foundation;
#import "CreatureUtil.h"
#import "Fightable.h"

@implementation CreatureUtil

+ (NSString *)standardArmorClassDescription {
    return @"Natural Armor";
}

/*
 * Counts the treasure hoard a creature guards.
 */
+ (NSUInteger)treasureHoardOf:(Creature *)creature {
    return [creature treasureHoardValue];
}

@end

@implementation Creature (Treasure)

- (NSUInteger)treasureHoardValue {
    return (NSUInteger)(self.hitPoints.max * 10);
}

@end

File: sample-projects/swift/Sources/CellarsAndCentaurs/Application/CreatureUtil.swift
import Foundation

enum CreatureUtil {
    static let STANDARD_ARMOR_CLASS_DESCRIPTION = "Natural Armor"

    /*
     Counts the treasure hoard a creature guards.
     */
    static func treasureHoard(of creature: Creature) -> Int {
        let hoard = (creature.hitPoints?.maximum ?? 0) * 10
        return hoard
    }

    static func describe(_ creatures: [Creature]) -> [String: Int] {
        Dictionary(grouping: creatures, by: { $0.type.rawValue }).mapValues { $0.count }
    }
}
